apps/api/src/persistence/repositories/repository.provider.ts

```diff
@@ -17,11 +19,16 @@ import mapArangoEdgeDocumentToEdgeConnectionDTO from '../database/utilities/mapA
 import mapDatabaseDTOToEntityDTO from '../database/utilities/mapDatabaseDTOToEntityDTO';
 import mapEdgeConnectionDTOToArangoEdgeDocument from '../database/utilities/mapEdgeConnectionDTOToArangoEdgeDocument';
 import mapEntityDTOToDatabaseDTO from '../database/utilities/mapEntityDTOToDatabaseDTO';
+import ArangoCategoryRepository from './ArangoCategoryRepository';
 import { RepositoryForEntity } from './repository-for-entity';
 
 @Injectable()
 export class RepositoryProvider
-    implements IRepositoryProvider, IEdgeConnectionRepositoryProvider, ITagRepositoryProvider
+    implements
+        IRepositoryProvider,
+        IEdgeConnectionRepositoryProvider,
+        ITagRepositoryProvider,
+        ICategoryRepositoryProvider
```

Comment:
What is the benefit of making each of these separate instead of having a single `IRepositoryProvider`? Do we inject to the minimum interface needed anywhere? I.e. if we only need tags, do we inject an `ITagRepositoryProvider`? This could leave open the possibility of implementing them in different classes (or even with different backing db's) which is kind of nice, I suppose.